spreadsheets.js

```diff
@@ -9,17 +9,94 @@ var EXTREMA_DEFAULTS = {
     'minute': 0,
     'warning_hours': 1,
 };
+var BASE_DATE_FORMATS = [
+    "YYYY-MM-DD",
+    "YYYY-MM-D",
+    "YYYY-M-DD",
+    "YYYY-M-D",
+    "YY-MM-DD",
+    "YY-MM-D",
+    "YY-M-DD",
+    "YY-M-D",
+    "MM-DD-YYYY",
+    "MM-D-YYYY",
+    "M-DD-YYYY",
+    "M-D-YYYY",
+    "MM-DD-YY",
+    "MM-D-YY",
+    "M-DD-YY",
+    "M-D-YY",
+    "YYYY-DD-MM",
+    "YYYY-D-MM",
+    "YYYY-DD-M",
+    "YYYY-D-M",
+    "YY-DD-MM",
+    "YY-D-MM",
+    "YY-DD-M",
+    "YY-D-M",
+    "DD-MM-YYYY",
+    "D-MM-YYYY",
+    "DD-M-YYYY",
+    "D-M-YYYY",
+    "DD-MM-YY",
+    "D-MM-YY",
+    "DD-M-YY",
+    "D-M-YY",
+    "YYYY-MMMM Do",
+    "YY-MMMM Do",
+    "MMMM Do YYYY",
+    "MMMM Do YY",
+    "YYYY-Do MMMM",
+    "YY-Do MMMM",
+    "Do MMMM YYYY",
+    "Do MMMM YY",
+    "YYYY-MMMM DD",
+    "YY-MMMM DD",
+    "MMMM DD YYYY",
+    "MMMM DD YY",
+    "YYYY-DD MMMM",
+    "YY-DD MMMM",
+    "DD MMMM YYYY",
+    "DD MMMM YY",
+    "YYYY-MMMM D",
+    "YY-MMMM D",
+    "MMMM D YYYY",
+    "MMMM D YY",
+    "YYYY-D MMMM",
+    "YY-D MMMM",
+    "D MMMM YYYY",
+    "D MMMM YY",
+];
+var BASE_TIME_FORMATS = [
+    "HH:mm",
+    "H:mm",
+    "H:m"
+];
 var DATE_FORMATS = [
-    "YYYYMMDD HHmm",
-    "HHmm DDMMYYYY"
 ];
+BASE_DATE_FORMATS.forEach(function(date_format) {
+    BASE_TIME_FORMATS.forEach(function(time_format) {
+        var date_format2 = date_format.replace(/[-]/g, '');
+        var time_format2 = time_format.replace(/[:]/g, '');
+        DATE_FORMATS.push(date_format + " " + time_format);
+        DATE_FORMATS.push(time_format + " " + date_format);
+        DATE_FORMATS.push(date_format2 + " " + time_format2);
+        DATE_FORMATS.push(time_format2 + " " + date_format2);
+        DATE_FORMATS.push(date_format2 + " " + time_format);
+        DATE_FORMATS.push(time_format + " " + date_format2);
+        DATE_FORMATS.push(date_format + " " + time_format2);
+        DATE_FORMATS.push(time_format2 + " " + date_format);
+    });
```

Comment:
Yes, i saw that. Not much you can do though. Unless we can order these in some way, brute force is out only option.